### ISSUE TITLE
Silo: Actions are not carried over into combined silos

### ISSUE THREAD

**Issue by Rocky28447** [Sleitnick/RbxUtil#182]
When combining silos together through `Silo.combine()`, the actions from the provided silos are not carried over into the combined silo. 

This may be the root cause of #97.

Example code:
```lua
local ReplicatedStorage = game:GetService("ReplicatedStorage")

local Packages = ReplicatedStorage.Packages
local Shared = ReplicatedStorage.Shared

local Silo = require(Packages.Silo)

local KillsFragmentInfo = require(Shared.Silos.KillsFragment)
local KillsFragment = Silo.new(KillsFragmentInfo.State, KillsFragmentInfo.Modifiers)

local DeathsFragmentInfo = require(Shared.Silos.DeathsFragment)
local DeathsFragment = Silo.new(DeathsFragmentInfo.State, DeathsFragmentInfo.Modifiers)

local CompleteFragment = Silo.combine({
    Kills = KillsFragment,
    Deaths = DeathsFragment,
})

print(KillsFragment)
print(DeathsFragment)
print(CompleteFragment)
```

Output:

![image](https://github.com/Sleitnick/RbxUtil/assets/26193486/a3efd83d-8c90-4614-8f77-3a0f14ab75d7)

**Comment by Sleitnick** [comment]
Here's some info about the issue:

1. You can still dispatch actions through the original silos and it will affect the combined silo.
1. Populating the combined silo with all the actions has the possibility of name conflicts.

Regarding 1, this is a temporary workaround.

Regarding 2, this makes the solution a bit complicated. While the internal modifier adds a prefix to avoid the naming conflict, the name of the action remains the same. Thus, with possible naming conflicts, either (a) an error should be thrown if a naming conflict occurs, or (b) a prefix should also be added to the action name (which I think would be confusing). I am leaning toward option (a). Just throw an error for naming conflicts.

**Comment by Sleitnick** [comment]
Went with option (a). Actions are combined, and an error is thrown if a name conflict is found. Published in Silo v0.2.0.
